internal/provider/finalize_version_upgrade_resource.go

```diff
@@ -0,0 +1,159 @@
+      "cluster_id": schema.StringAttribute{
+        Required: true,
+        PlanModifiers: []planmodifier.String{
+          stringplanmodifier.RequiresReplace(),
+        },
+      },
+      "cockroach_version": schema.StringAttribute{
+        Required: true,
+        PlanModifiers: []planmodifier.String{
+          stringplanmodifier.RequiresReplace(),
+        },
+      },
+      "id": schema.StringAttribute{
+        Computed: true,
+        PlanModifiers: []planmodifier.String{
+          stringplanmodifier.UseStateForUnknown(),
+        },
+      },
+    },
+  }
+}
+
+func (r *finalizeVersionUpgradeResource) Metadata(
+  _ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse,
+) {
+  resp.TypeName = req.ProviderTypeName + "_finalize_version_upgrade"
+}
+
+func (r *finalizeVersionUpgradeResource) Configure(
+  _ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse,
+) {
+  if req.ProviderData == nil {
+    return
+  }
+  var ok bool
+  if r.provider, ok = req.ProviderData.(*provider); !ok {
+    resp.Diagnostics.AddError("Internal provider error",
+      fmt.Sprintf("Error in Configure: expected %T but got %T", provider{}, req.ProviderData))
+  }
+}
+
+func (r *finalizeVersionUpgradeResource) Create(
+  ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse,
+) {
+  if r.provider == nil || !r.provider.configured {
+    addConfigureProviderErr(&resp.Diagnostics)
+    return
+  }
+  var plan FinalizeVersionUpgrade
+  diags := req.Plan.Get(ctx, &plan)
+  resp.Diagnostics.Append(diags...)
+  if resp.Diagnostics.HasError() {
+    return
+  }
+  clusterID := plan.ClusterId.ValueString()
+
+  clusterObj, _, err := r.provider.service.GetCluster(ctx, clusterID)
+  if err != nil {
+    resp.Diagnostics.AddError("Error retrieving cluster info", formatAPIErrorMessage(err))
+    return
+  }
+
+  if clusterObj.UpgradeStatus != client.CLUSTERUPGRADESTATUS_PENDING_FINALIZATION {
+    resp.Diagnostics.AddError(
+      "Cluster is not pending upgrade finalization",
+      fmt.Sprintf("Expected upgrade status '%s', currently '%s'", client.CLUSTERUPGRADESTATUS_PENDING_FINALIZATION, clusterObj.UpgradeStatus),
+    )
+  }
+
+  pendingMajorVersion := strings.Join(strings.Split(clusterObj.CockroachVersion, ".")[:2], ".")
+  if pendingMajorVersion != plan.CockroachVersion.ValueString() {
+    resp.Diagnostics.AddAttributeError(
+      path.Root("cockroach_version"),
+      "Unexpected version",
+      fmt.Sprintf("Set to '%s' to finalize the pending upgrade.", pendingMajorVersion))
+    return
+  }
+
+  finalizedStatus := client.CLUSTERUPGRADESTATUS_FINALIZED
+  if _, _, err = r.provider.service.UpdateCluster(ctx, clusterID, &client.UpdateClusterSpecification{UpgradeStatus: &finalizedStatus}); err != nil {
+    resp.Diagnostics.AddError("Error finalizing cluster upgrade", formatAPIErrorMessage(err))
+    return
+  }
+
+  plan.ID = plan.ClusterId
```

Comment:
Could we use `ID` with `MarkdownDescription: "Cluster ID",` instead of having both `id` and `cluster_id` ?